src/values/AbstractObjectValue.js

```diff
@@ -486,10 +486,16 @@ export default class AbstractObjectValue extends AbstractValue {
       let generateAbstractGet = () => {
         let type = Value;
         if (P === "length" && Value.isTypeCompatibleWith(this.getType(), ArrayValue)) type = NumberValue;
+        let unwrapped = this;
+        if (this.kind === "explicit conversion to object") {
+          // We can unwrap because the generated property access
+          // will implicitly convert it to an object.
+          unwrapped = this.args[0];
+        }
         return AbstractValue.createTemporalFromBuildFunction(
           this.$Realm,
           type,
-          [this],
+          [unwrapped],
```

Comment:
We only want to use `wrapped` in the above case, otherwise we should still use `this`?

Reply:
Yes, I initialized it to `this`